### PR TITLE
add loss option for TCNForecaster

### PR DESCRIPTION
**Pull request by TheaperDeng**
1. Add a `PYTORCH_REGRESSION_LOSS_MAP` for future use in built-in model
2. add loss option for TCNForecaster

### CODE REVIEW

File: pyzoo/zoo/zouwu/model/forecast/model/tcn.py [hunk 1]
```diff
@@ -146,7 +147,12 @@ def optimizer_creator(model, config):
 
 
 def loss_creator(config):
-    return nn.MSELoss()
+    loss_name = config.get("loss", "mse")
+    if loss_name in PYTORCH_REGRESSION_LOSS_MAP:
+        loss_name = PYTORCH_REGRESSION_LOSS_MAP[loss_name]
+    else:
+        raise RuntimeError("You need to provide valid loss name within mse, mae and huber_loss")
```
Comment (yushan111) [inline pyzoo/zoo/zouwu/model/forecast/model/tcn.py:154]: Got {loss_name}.

Comment (yushan111) [inline pyzoo/zoo/zouwu/model/forecast/model/tcn.py:154]: Shall we move loss_creator into pytorch_base_model? It seems to be commonly used.

**Review by yushan111 — COMMENTED** [review]
Others LGTM

**Comment by TheaperDeng** [comment]
PR validation combine pass:
http://10.239.47.210:18888/job/ZOO-PR-Python-ExampleTests/2518/
http://10.239.47.210:18888/job/ZOO-PR-Validation/5288/

**Pull request merged**